src/v/cloud_storage/access_time_tracker.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage/access_time_tracker.h"
+#include "serde/serde.h"
+#include "units.h"
+
+#include <seastar/core/smp.hh>
+#include <seastar/core/coroutine.hh>
+#include <absl/container/btree_map.h>
+#include <exception>
+#include <variant>
+
+namespace absl {
+
+template<class Key, class Value>
+void write(iobuf& out,  const btree_map<Key, Value>& btree) {
+    using serde::write;
+    write(out, static_cast<uint64_t>(btree.size()));
+}
+
+template<class Key, class Value>
+void read_nested(iobuf_parser& in, btree_map<Key, Value>& , size_t const bytes_left_limit) {
+    using serde::read_nested;
+    uint64_t sz;
+    read_nested(in, sz, bytes_left_limit);
+}
+
+}
+
+namespace cloud_storage {
+
+void access_time_tracker::add_timestamp(
+    std::string_view key, std::chrono::system_clock::time_point ts) {
+    uint32_t seconds = std::chrono::time_point_cast<std::chrono::seconds>(ts)
+                     .time_since_epoch()
+                     .count();
+    uint32_t hash = xxhash_32(key.data(), key.size());
+    _table.data[hash] = seconds; 
```

Comment:
Just curious, why you chose `k=1` (single row, single hash function) for the count-min sketch? Might be worth adding a comment?